### ISSUE TITLE
Open in Merge Editor button shall always remove the conflict markers

### ISSUE THREAD

**Issue by sandy081** [microsoft/vscode#162191]
Testing #161833

- Create conflicts
- Open conflicts file
- Open in Merge editor (file becomes dirty)
- Close the conflicts file (do not save)
- Open the conflicts file again
- Open in Merge editor
🐛 Conflict markers are not removed. I would expect Merge editor opens as in 3rd step

**Comment by hediet** [comment]
Step 4 will restore the file, thus effectively re-introduce the conflict markers.
Step 6 will just reveal the merge editor and not remove the conflict markers, as the merge editor is still opened.

@bpasero @jrieken ideas? :)

**Comment by bpasero** [comment]
Yeah, I wonder if opening the merge editor should simply close the textual merge editor. We already save it (as far as I know), so we could also close it. In other words, the 3-way merge editor would replace the text merge editor. Note that this is only possible if the text merge editor is not dirty, so a bit of proper `async` code is needed, but should work.

Obviously people could still trigger this, but maybe less likely.

**Comment by jrieken** [comment]
> I wonder if opening the merge editor should simply close the textual merge editor.

 😨 I fear rage when we take the text editor away, esp when folks do experimental-clicking

> Step 4 will restore the file, thus effectively re-introduce the conflict markers.

@hediet Can't we detect that the file has changed outside of the 3wm-editor and handle that special? Like re-start merge in the case of external modifications? That is maybe a little drastic but I believe the scenario is a corner-case and restarting isn't the worst

**Comment by bpasero** [comment]
(this was brought up in the standup): I can also investigate if we can prevent the dialog from asking given the file is still opened in the merge editor. though it would probably not fix all the issues, because a user could still "File > Revert".

maybe the merge editor could detect this and show a in-editor notification with an option to re-merge and remove the markers again.

**Comment by hediet** [comment]
both ideas sound good!
